doc/syntax.bnf

```diff
@@ -112,6 +113,10 @@ reserved "why3"
 
 // TODO rule naming, positive / negative
 
+// Unification hint
+<hint> ::=
+  | <patt> "≡" <patt> "↪" <pvar> "≡" <term> {"," <term> "≡" <term>}*
```

Comment:
Why do you restrict to <patt> and <pvar>? Why not using <term> every where?

Reply:
I'm afraid I don't understand the question.

Reply:
Where <hint> is used is not defined. It should be part of <command>:
```
| set "unif" <term> "≡" <term> "↪" <term> "≡" <term> {"," <term> "≡" <term>}*
```

Reply:
You can reject unification rules that cannot be encoded into a rewrite rule at scoping time.

Reply:
I prefer using set "unif_rule" for not adding yet another reserved word in the grammar, and also because it is just a way to modify the unification algorithm.

Reply:
Ah, I see now why you want to have a `<patt>`. Note however that, in parser.ml, a rule is parsed as term*term. LHS that are not patterns are rejected at scoping time. But I don't understand why you have `<pvar>`.

Reply:
The form of a hint is always `?x == term`. There is an error in my line, there shouldn't be `<term> == <term>`.